shared/gh/js/views/gh.delete-module.js

```diff
@@ -0,0 +1,350 @@
+                    // Call itself when more series need to be fetched
+                    } else {
+                        deleteSeries(seriesToDelete[done], callback);
+                    }
+                });
+            // If the series isn't borrowed into any other modules and was not originally created
+            // in this module only unlink it from the series
+            } else if (series && series.id && !_.contains(templateData.borrowedElsewhere, series.id)) {
+                // Remove the series from the module
+                orgUnitAPI.deleteOrgUnitSeries(moduleId, series.id, function(err) {
+                    done++;
+                    // When we're done, execute the callback
+                    if (todo === done) {
+                        callback(null);
+                    // Call itself when more series need to be fetched
+                    } else {
+                        deleteSeries(seriesToDelete[done], callback);
+                    }
+                });
+            } else {
+                callback(null);
+            }
+        };
+
+        // Get all series inside of the module
+        orgUnitAPI.getOrgUnit(moduleId, true, function(err, moduleOrgUnit) {
+            // Keep track of how much work we have to do
+            todo = moduleOrgUnit.Series.length;
+            seriesToDelete = moduleOrgUnit.Series;
+
+            // Delete the series
+            deleteSeries(seriesToDelete[0], function() {
+                // Delete the module
+                orgUnitAPI.deleteOrgUnit(moduleId, function(err) {
+                    if (err) {
+                        return gh.utils.notification('Could not delete the module', constants.messaging.default.error, 'error');
+                    }
+
+                    // Hide the modal
+                    $('#gh-delete-module-modal').modal('hide');
+
+                    // Show a success notification after the module and all its series have been deleted
+                    gh.utils.notification('The module has been removed', null, 'success');
+
+                    // Remove the module from the navigation
+                    $('.list-group-item[data-id="' + moduleId + '"]').remove();
+
+                    // Remove the module and series from the state
+                    gh.utils.removeFromState(['series', 'module']);
+                });
+            });
+        });
+
+        // Avoid default form submit behaviour
+        return false;
+    };
+
+
+    ////////////////////
+    // INITIALISATION //
+    ////////////////////
+
+    /**
+     * Render and show the 'delete module' modal dialog
+     *
+     * @private
+     */
+    var setUpDeleteModuleModal = function() {
+        // Cache the module ID
+        moduleId = parseInt($(this).attr('data-id'), 10);
+
+        // Render the modal
+        gh.utils.renderTemplate($('#gh-delete-module-modal-template'), {'data': null}, $('#gh-delete-module-modal-container'));
```

Comment:
We can probably just pass in `null` instead of `{'data': null}`. (see previous comment)